common/common.go

```diff
@@ -79,16 +80,18 @@ func InitHoneycombFromEnvVars() error {
 		}
 	}
 
-	apiHost := os.Getenv("API_HOST")
+	apiHost = os.Getenv("API_HOST")
 	if apiHost == "" {
 		apiHost = "https://api.honeycomb.io"
 	}
 
-	dataset := os.Getenv("DATASET")
+	dataset = os.Getenv("DATASET")
 	if dataset == "" {
 		dataset = "honeycomb-cloudwatch-logs"
 	}
 
+	errorDataset = os.Getenv("ERROR_DATASET")
```

Comment:
What would you think about sending errors to the normal dataset if `ERROR_DATASET` isn't set?

If there were parsing errors I didn't expect, I think I'd rather see them in my normal dataset, and then decide to filter them out if necessary, rather than have them silently dropped if I forgot to configure `ERROR_DATASET`.

Reply:
I was leaning toward an opt-in solution so that people who upgrade their integration don't get new behavior that they don't expect. Also some of the errors are unavoidable: for example, lambda writes three lines by itself for every single invocation - that would be three events the customer is getting charged for.

```
START RequestId: 27c12e28-5944-11e8-b012-334998af0607 Version: $LATEST
END RequestId: 27c12e28-5944-11e8-b012-334998af0607
REPORT RequestId: 27c12e28-5944-11e8-b012-334998af0607	Duration: 258.59 ms	Billed Duration: 300 ms Memory Size: 128 MB	Max Memory Used: 31 MB	
```

(maybe we should have a specialized Lambda handler that can filter or at least make use of this information)

Reply:
:+1: makes sense, thanks for the explanation